site/docs/backward-compatibility.md

```diff
@@ -0,0 +1,82 @@
+---
+layout: documentation
+title: Backward Compatibility
+---
+
+# Backward Compatibility
+
+Bazel is evolving, and we will make changes to Bazel that at times will be
+incompatible and require some changes from Bazel users.
+
+## At a glance
+
+1. Every breaking change is guarded with an `--incompatible_*` flag.
+1. Newly introduced incompatible flags default to off.
+1. For every `--incompatible_*` flag we have a github issue explaining
+   the change in behavior and giving a migration recipe.
+1. The migration window is at least one release long and is set by the author of the incompatible change.
+1. We announce what set of flags we flip with the next release one release in advance.
+1. APIs and behavior guarded by an `--experimental_*` flag can change at any time.
+1. Users should never run their production builds with `--experimental_*`  or `--incompatible_*` flags.
+
+## What is stable functionality?
+
+In general, if an API or a behavior is available in Bazel without
+`--experimental_...` flag, we consider it a stable, supported feature.
+This includes:
+
+* Starlark language and APIs
+* Rules bundled with Bazel
+* Bazel APIs such as Remote Execution APIs or Build Event Protocol
+* Flags and their semantics
+
+## Incompatible changes and migration recipes
+
+When we introduce an incompatible change, we try to make it easier for Bazel
+users to update their code. We do this by means of _migration windows_ and
+_migration recipes_.
+
+Migration window is one or more release of Bazel during which a migration from
+old functionality to new functionality is possible, according to a migration
+recipe.
+
+During the migration window, both the old functionality and the new functionality
```

Comment:
Could we add something for the case where a migration recipe is not available? For example, the [aapt2 migration](https://github.com/bazelbuild/bazel/issues/6907) is not doable mechanically most of the time, but we keep another flag around to switch to the old behavior.